### PR TITLE
CI staging namespace

### PR DESCRIPTION
**Pull request by bbguimaraes**
Secret mirroring is still missing.  For intended usage, see:

[8c3560f](https://github.com/openshift/release/commit/8c3560ff25b16516b2d85d017018824aa2df0227)

### CODE REVIEW

File: hack/mkpjpod.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# - PULL_AUTHOR
+set -euo pipefail
+
+if [[ "$#" -ne 1 ]]; then
+    echo >&2 "Usage: $0 job_name"
+    exit 1
+fi
+job=$1
+img=registry.svc.ci.openshift.org/ci/test-infra:binaries
+podman run --rm -iv "$PWD:/tmp/release:Z" -w /tmp/release "$img" bash <<-EOF
```
Comment (petr-muller) [inline hack/mkpjpod.sh]: Hm are we that confident everyone is running `podman` already?

File: cluster/ci/config/prow/plugins.yaml [hunk 2]
```diff
@@ -403,55 +403,130 @@ config_updater:
 
     cluster/test-deploy/gcp/*.yaml:
       name: cluster-profile-gcp
+    cluster/test-deploy/gcp/*.yaml:
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/plugins.yaml]: Maybe we should change the upload plugin to allow a list of namespaces?

File: hack/mkpjpod.sh (hunk 1 above)
Comment (stevekuznetsov) [reply hack/mkpjpod.sh]: I'm not :)

Comment (stevekuznetsov) [reply hack/mkpjpod.sh]: Also the `:Z` should not be necessary, `:z` should suffice

Comment (bbguimaraes) [reply hack/mkpjpod.sh]: > Hm are we that confident everyone is running podman already?

Can we make them? =)

> I'm not :)

Shame on you.

> Also the :Z should not be necessary, :z should suffice

I always think of it as `z` == "security", `Z` == "more security". I'll have to find a better mnemonic (or RTFM).

**Comment by bbguimaraes** [comment]
Updated:

- rebased
- added the `ci-admins` group as admins in the namespace
- added secret mirroring configuration

I have deployed the secret mirroring controller and everything in this PR on `api.ci` to test the deployment configuration. Namespace `ci-stg` is live and secrets are being mirrored. I have left out:

1. `ci-operator` service account
2. configmaps
3. `config_updater` configuration
3. rolebinding that gives regular users access

I will do (1) and apply (2) manually tomorrow and do a full test, then maybe tackle the `config_updater` configuration change.

File: cluster/ci/config/prow/openshift/ci-operator/stage.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+kind: List
+apiVersion: v1
+items:
+- kind: Role
+  apiVersion: authorization.openshift.io/v1
+  metadata:
+    name: ci-stg-user
+  rules:
```
Comment (stevekuznetsov) [inline cluster/ci/config/prow/openshift/ci-operator/stage.yaml]: Do we need create on events here, too?

Comment (bbguimaraes) [reply cluster/ci/config/prow/openshift/ci-operator/stage.yaml]: `ci-operator` will use its own service account. This role is for the individual user accounts, which should interface with the namespace exclusively by creating pods generated by `mkpod`.

The reason ConfigMaps are included is to allow template authors (or adventurous users) to create their own template CM and change the job to use it (which have to has to be in the same namespace per kube's restrictions).

Comment (stevekuznetsov) [reply cluster/ci/config/prow/openshift/ci-operator/stage.yaml]: Of course, that makes sense

**Pull request merged**